pandas/tests/reshape/concat/test_concat.py

```diff
@@ -348,10 +348,44 @@ def test_concat_single_with_key(self):
         expected = concat([df, df], keys=["foo", "bar"])
         tm.assert_frame_equal(result, expected[:10])
 
-    def test_concat_no_items_raises(self):
-        with pytest.raises(ValueError, match="No objects to concatenate"):
+    def test_concat_no_items_and_initializer_raises(self):
+        with pytest.raises(
+            ValueError, match="No objects to concatenate and no initializer not set"
```

Comment:
"No objects to concatenate and no initializer set"

Reply:
> I am not sure whether this approach is reasonable, but code-wise it looks good to me.
> Some comments below.

Thanks! I'll amend my code accordingly -- it's just an approach to the issue described in #40820 that I could think of at the moment -- would like to hear more from others if this approach is reasonable too.

Thanks so much for the review